sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -455,9 +455,10 @@ public Query apply(Function input, Context context) {
 
                 String columnName = reference.info().ident().columnIdent().fqn();
                 QueryBuilderHelper builderHelper = QueryBuilderHelper.forType(reference.valueType());
-                return new FilteredQuery(
-                        Queries.newMatchAllQuery(),
-                        new NotFilter(builderHelper.rangeFilter(columnName, null, null, true, true)));
```

Comment:
check if NotFilter  was the problem with deleted docs and if we use it somewhere else